File: scripts/connfu-selenium.py
from selenium import selenium
import unittest, time, re

class connfu-selenium(unittest.TestCase):
    def setUp(self):
        self.verificationErrors = []
        self.selenium = selenium("localhost", 4444, "*chrome", "http://localhost:3000/")
        self.selenium.start()
    
    def test_connfu-selenium(self):
        sel = self.selenium
        sel.open("/")
        sel.wait_for_page_to_load("60000")
        sel.open("/")
        sel.click("link=Sign up")
        sel.wait_for_page_to_load("60000")
        sel.type("xpath=//form[@id='user_new']/dl[1]/dd/input", "jmeter7")
        sel.type("id=user_email", "abc@aol.com")
        sel.type("xpath=//form[@id='user_new']/dl[3]/dd/input", "jmeter7")
        sel.type("id=user_password_confirmation", "jmeter7")
        sel.click("id=user_submit")
        sel.wait_for_page_to_load("60000")
        sel.click("link=How it Works")
        sel.click("link=My apps")
        sel.wait_for_page_to_load("60000")
        sel.click("id=step_3")
        sel.wait_for_page_to_load("60000")
        sel.type("id=app_name", "jmeter7-app")
        sel.type("id=app_description", "This is jmeter7 connfu app")
        sel.click("id=app_submit")
        sel.wait_for_page_to_load("60000")
        sel.click("xpath=//div[@class='table']/a/div/div[2]/p/b")
        sel.wait_for_page_to_load("60000")
        sel.click("xpath=//div[@id='content']/div[3]")
        sel.click("xpath=//div[@id='content']/div[4]/div[2]/div")
        sel.click("link=Add channel")
        sel.select("id=channel_type", "RSS")
        sel.type("id=rss_uid", "twitter-uid")
        sel.type("id=rss_uri", "")
        sel.type("id=rss_uid", "rss-uid")
        sel.type("id=rss_uri", "http://www.bbc.co.uk")
        sel.click("id=rss_submit")
        sel.wait_for_page_to_load("60000")
        sel.click("id=app_usage_button")
        sel.click("link=Close")
        sel.click("link=Add channel")
        sel.select("id=channel_type", "Twitter")
        sel.type("id=twitter_uid", "twitter-uid")
        sel.type("id=add_twitter_input", "chiflendotcom")
        sel.type("id=add_twitter_filter_input", "api")
        sel.click("xpath=//div[@class='account_filters']/a/img")
        sel.click("xpath=//td[@class='button']/a/img")
        sel.click("id=create_twitter_channel_button")
        sel.wait_for_page_to_load("60000")
        sel.go_back()
        sel.wait_for_page_to_load("60000")
        sel.click("link=Add channel")
        sel.select("id=channel_type", "Voice")
        sel.type("id=voice_uid", "voice-uid")
        sel.select("id=voice_country", "United Kingdom")
        sel.type("id=voice_topic", "Conference powered by connFu.")
        sel.type("id=voice_welcome_message", "Welcome to connFu, you are going to join the conference")
        sel.type("id=voice_rejected_message", "You are not allowed to join this conference")
        sel.click("xpath=//div[@id='voice_form']//a[.='Cancel']")
        sel.click("link=Add channel")
        sel.select("id=channel_type", "Voice")
        sel.click("id=voice_topic")
        sel.type("id=voice_welcome_message", "Welcome to connFu, you are going to join the conference")
        sel.type("id=voice_rejected_message", "You are not allowed to join this conference")
        sel.click("id=voice_submit")
        sel.wait_for_page_to_load("60000")
        sel.click("xpath=//div[@id='content']//div[normalize-space(.)='Voice']")
        sel.type("id=whitelisted_name", "jmeter7")
        sel.type("id=whitelisted_phone", "11340927863")
        sel.click("xpath=//div[@id='whitelist_form']//button[.='Add']")
        sel.click("xpath=//div[@class='app-create-button']/input")
        sel.wait_for_page_to_load("60000")
        sel.click("xpath=//div[@id='content']//span[normalize-space(.)='http://www.bbc.co.uk']")
        sel.type("xpath=//form[@id='edit_rss_4eb3d8fc9c7ec109ba000037']/dl/div/dd/input", "http://www.cnn.com")
        sel.click("xpath=//div[@class='app-create-button']/input")
        sel.wait_for_page_to_load("60000")
        sel.click("link=Documentation")
        sel.wait_for_page_to_load("60000")
        sel.click("link=Support")
        sel.wait_for_page_to_load("60000")
        sel.click("link=How it Works")
        sel.wait_for_page_to_load("60000")
        sel.click("link=My apps")
        sel.wait_for_page_to_load("60000")
        sel.click("id=step_2")
        sel.wait_for_page_to_load("60000")
        sel.click("link=Support")
        sel.wait_for_page_to_load("60000")
        sel.click("link=My apps")
        sel.wait_for_page_to_load("60000")
        sel.click("id=step_1")
        sel.wait_for_page_to_load("60000")
        sel.click("link=jmeter7")
        sel.wait_for_page_to_load("60000")
        sel.type("id=user_password_confirmation", "jmeter7")
        sel.type("id=user_current_password", "jmeter7")
        sel.type("id=user_password", "jmeter7")
        sel.click("id=user_submit")
        sel.wait_for_page_to_load("60000")
        sel.click("link=Sign out")
        sel.wait_for_page_to_load("60000")
    
    def tearDown(self):
        self.selenium.stop()
        self.assertEqual([], self.verificationErrors)

if __name__ == "__main__":
    unittest.main()

File: scripts/connfu-selenium.php
<?php

require_once 'PHPUnit/Extensions/SeleniumTestCase.php';

class Example extends PHPUnit_Extensions_SeleniumTestCase
{
  protected function setUp()
  {
    $this->setBrowser("*chrome");
    $this->setBrowserUrl("http://localhost:3000/");
  }

  public function testMyTestCase()
  {
    $this->open("/");
    $this->waitForPageToLoad("60000");
    $this->open("/");
    $this->click("link=Sign up");
    $this->waitForPageToLoad("60000");
    $this->type("xpath=//form[@id='user_new']/dl[1]/dd/input", "jmeter7");
    $this->type("id=user_email", "abc@aol.com");
    $this->type("xpath=//form[@id='user_new']/dl[3]/dd/input", "jmeter7");
    $this->type("id=user_password_confirmation", "jmeter7");
    $this->click("id=user_submit");
    $this->waitForPageToLoad("60000");
    $this->click("link=How it Works");
    $this->click("link=My apps");
    $this->waitForPageToLoad("60000");
    $this->click("id=step_3");
    $this->waitForPageToLoad("60000");
    $this->type("id=app_name", "jmeter7-app");
    $this->type("id=app_description", "This is jmeter7 connfu app");
    $this->click("id=app_submit");
    $this->waitForPageToLoad("60000");
    $this->click("xpath=//div[@class='table']/a/div/div[2]/p/b");
    $this->waitForPageToLoad("60000");
    $this->click("xpath=//div[@id='content']/div[3]");
    $this->click("xpath=//div[@id='content']/div[4]/div[2]/div");
    $this->click("link=Add channel");
    $this->select("id=channel_type", "RSS");
    $this->type("id=rss_uid", "twitter-uid");
    $this->type("id=rss_uri", "");
    $this->type("id=rss_uid", "rss-uid");
    $this->type("id=rss_uri", "http://www.bbc.co.uk");
    $this->click("id=rss_submit");
    $this->waitForPageToLoad("60000");
    $this->click("id=app_usage_button");
    $this->click("link=Close");
    $this->click("link=Add channel");
    $this->select("id=channel_type", "Twitter");
    $this->type("id=twitter_uid", "twitter-uid");
    $this->type("id=add_twitter_input", "chiflendotcom");
    $this->type("id=add_twitter_filter_input", "api");
    $this->click("xpath=//div[@class='account_filters']/a/img");
    $this->click("xpath=//td[@class='button']/a/img");
    $this->click("id=create_twitter_channel_button");
    $this->waitForPageToLoad("60000");
    $this->goBack();
    $this->waitForPageToLoad("60000");
    $this->click("link=Add channel");
    $this->select("id=channel_type", "Voice");
    $this->type("id=voice_uid", "voice-uid");
    $this->select("id=voice_country", "United Kingdom");
    $this->type("id=voice_topic", "Conference powered by connFu.");
    $this->type("id=voice_welcome_message", "Welcome to connFu, you are going to join the conference");
    $this->type("id=voice_rejected_message", "You are not allowed to join this conference");
    $this->click("xpath=//div[@id='voice_form']//a[.='Cancel']");
    $this->click("link=Add channel");
    $this->select("id=channel_type", "Voice");
    $this->click("id=voice_topic");
    $this->type("id=voice_welcome_message", "Welcome to connFu, you are going to join the conference");
    $this->type("id=voice_rejected_message", "You are not allowed to join this conference");
    $this->click("id=voice_submit");
    $this->waitForPageToLoad("60000");
    $this->click("xpath=//div[@id='content']//div[normalize-space(.)='Voice']");
    $this->type("id=whitelisted_name", "jmeter7");
    $this->type("id=whitelisted_phone", "11340927863");
    $this->click("xpath=//div[@id='whitelist_form']//button[.='Add']");
    $this->click("xpath=//div[@class='app-create-button']/input");
    $this->waitForPageToLoad("60000");
    $this->click("xpath=//div[@id='content']//span[normalize-space(.)='http://www.bbc.co.uk']");
    $this->type("xpath=//form[@id='edit_rss_4eb3d8fc9c7ec109ba000037']/dl/div/dd/input", "http://www.cnn.com");
    $this->click("xpath=//div[@class='app-create-button']/input");
    $this->waitForPageToLoad("60000");
    $this->click("link=Documentation");
    $this->waitForPageToLoad("60000");
    $this->click("link=Support");
    $this->waitForPageToLoad("60000");
    $this->click("link=How it Works");
    $this->waitForPageToLoad("60000");
    $this->click("link=My apps");
    $this->waitForPageToLoad("60000");
    $this->click("id=step_2");
    $this->waitForPageToLoad("60000");
    $this->click("link=Support");
    $this->waitForPageToLoad("60000");
    $this->click("link=My apps");
    $this->waitForPageToLoad("60000");
    $this->click("id=step_1");
    $this->waitForPageToLoad("60000");
    $this->click("link=jmeter7");
    $this->waitForPageToLoad("60000");
    $this->type("id=user_password_confirmation", "jmeter7");
    $this->type("id=user_current_password", "jmeter7");
    $this->type("id=user_password", "jmeter7");
    $this->click("id=user_submit");
    $this->waitForPageToLoad("60000");
    $this->click("link=Sign out");
    $this->waitForPageToLoad("60000");
  }
}
?>